### PR TITLE
docs: add workflow to auto update docs

### PR DESCRIPTION
**Pull request by charlieegan3**

### CODE REVIEW

File: .github/workflows/update-docs.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+          private_key: ${{ secrets.PLATFORM_AUTOMATION_GH_APP_PEM_KEY }}
+
+      - name: Trigger Update in docs repo
+        run: |
+          curl -X "POST" "https://api.github.com/repos/StyraInc/docs/actions/workflows/update-regal.yaml/dispatches" \
+            -H 'Accept: application/vnd.github+json' \
+            -H 'Authorization: Bearer ${{ steps.generate_token.outputs.token }}' \
+            -H 'X-Github-Api-Version: 2022-11-28' \
+            -H 'Content-Type: application/json; charset=utf-8' \
+            -d $'{
```
Comment (srenatus) [inline .github/workflows/update-docs.yaml:34]: What does the `$` do here? 🤔

Comment (charlieegan3) [reply .github/workflows/update-docs.yaml:34]: I actually think in this case it's not doing anything, but if there were `\n`s in there, they'd be handled as new lines. I think, just a bash thing as I understand it.

**Review by anderseknert — APPROVED** [review]

File: .github/workflows/update-docs.yaml (hunk 1 above)
Comment (charlieegan3) [reply .github/workflows/update-docs.yaml]: I have removed it, it's not needed. It's just set in the generated commands from my REST gui thingy.

Comment (srenatus) [reply .github/workflows/update-docs.yaml]: Interesting! I'll have to read up on that.

**Pull request merged**